scripts/post_install_script.sh

```diff
@@ -211,26 +250,170 @@ modules:
+      version: [2.4.0]
+    munge:
+      # Refer to ParallelCluster global munge space
+      variants: localstatedir=/var
+    openmpi:
+      variants: fabrics=ofi +pmi +legacylaunchers schedulers=slurm
+      version: [4.1.0,4.0.3]
+    intel-mpi:
+      version: [2020.2.254]
+    slurm:
+      variants: +pmix sysconfdir=/opt/slurm/etc
+      version: [20-02-4-1]
+    libfabric:
+      variants: fabrics=efa,tcp,udp,sockets,verbs,shm,mrail,rxd,rxm
+      version: [1.11.1,1.9.1]
+    mpich:
+      # For EFA (requires ch4)
+      variants: ~wrapperrpath pmi=pmi netmod=ofi device=ch4
+    libevent:
+      version: [2.1.8]
+    openblas:
+      version: [0.3.10]
+
+  modules:
+    enable:
+      - tcl
+    prefix_inspections:
+      bin:
+        - PATH
+      man:
+        - MANPATH
+      share/man:
+        - MANPATH
+      share/aclocal:
+        - ACLOCAL_PATH
+      lib:
+        - LIBRARY_PATH
+      lib64:
+        - LIBRARY_PATH
+      include:
+        - CPATH
+      lib/pkgconfig:
+        - PKG_CONFIG_PATH
+      lib64/pkgconfig:
+        - PKG_CONFIG_PATH
+      share/pkgconfig:
+        - PKG_CONFIG_PATH
+      '':
+        - CMAKE_PREFIX_PATH
+    tcl:
+      verbose: True
+      hash_length: 6
+      projections:
+        all: '{name}/{version}-{compiler.name}-{compiler.version}'
+        ^libfabric: '{name}/{version}-{^mpi.name}-{^mpi.version}-{^libfabric.name}-{^libfabric.version}-{compiler.name}-{compiler.version}'
+        ^mpi: '{name}/{version}-{^mpi.name}-{^mpi.version}-{compiler.name}-{compiler.version}'
+      whitelist:
+        - gcc
+      blacklist:
+        - slurm
+      all:
+        conflict:
+          - '{name}'
+        suffixes:
+          '^openblas': openblas
+          '^netlib-lapack': netlib
+        filter:
+          environment_blacklist: ['CPATH', 'LIBRARY_PATH']
+        environment:
+          set:
+            '{name}_ROOT': '{prefix}'
+        autoload:  direct
+      gcc:
+        environment:
+          set:
+            CC: gcc
+            CXX: g++
+            FC: gfortran
+            F90: gfortran
+            F77: gfortran
+      openmpi:
+        environment:
+          set:
+            SLURM_MPI_TYPE: "pmix"
+            OMPI_MCA_btl_tcp_if_exclude: "lo,docker0,virbr0"
+      miniconda3:
+        environment:
+          set:
+            CONDA_PKGS_DIRS: ~/.conda/pkgs
+            CONDA_ENVS_PATH: ~/.conda/envs
+    lmod:
+      hierarchy:
+        - mpi
+EOF
+	cat ${spack_install_path}/var/spack/environments/aws/spack.yaml
+
     echo "OSUSER=${OSUSER}"
     echo "OSGROUP=${OSGROUP}"
 	chown -R ${OSUSER}:${OSGROUP} ${spack_install_path}
     chmod -R go+rwX ${spack_install_path}
 
     #. /etc/profile.d/spack.sh
-	su - ${OSUSER} -c ". /etc/profile && spack bootstrap"
+    su - ${OSUSER} -c ". /etc/profile && curl -o /tmp/amzn2-e4s.pub https://s3.amazonaws.com/spack-mirrors/amzn2-e4s/build_cache/_pgp/7D344E2992071B0AAAE1EDB0E68DE2A80314303D.pub && spack gpg trust /tmp/amzn2-e4s.pub"
```

Comment:
parameterize the keypath

Reply:
Parameterizing anything in the post_install_script is much harder than config.ini. Will come back to this.

Reply:
I'm going to leave this hardcoded until we see a change on the cache key.